File: training-data/elixir/guard_clauses_demo.exs
defmodule GuardClauses do
  def classify(n) when is_integer(n) and n < 0, do: :negative
  def classify(0), do: :zero
  def classify(n) when is_integer(n) and rem(n, 2) == 0, do: :even
  def classify(n) when is_integer(n), do: :odd
  def classify(_), do: :not_a_number
end

IO.inspect(GuardClauses.classify(-5))
IO.inspect(GuardClauses.classify(0))
IO.inspect(GuardClauses.classify(4))
IO.inspect(GuardClauses.classify(7))
IO.inspect(GuardClauses.classify("x"))

File: training-data/erlang/guard_clauses_demo.erl
-module(guard_clauses_demo).
-export([classify/1]).

classify(N) when is_integer(N), N < 0 -> negative;
classify(0) -> zero;
classify(N) when is_integer(N), N rem 2 =:= 0 -> even;
classify(N) when is_integer(N) -> odd;
classify(_) -> not_a_number.
